internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,713 @@
+	outputs := make([]string, 0)
+	for _, site := range input {
+		outputs = append(outputs, site.Id)
+	}
+	return outputs
+}
+
+func expandPacketCoreControlLocalDiagnosticsAccessConfiguration(input []LocalDiagnosticsAccessConfigurationModel) packetcorecontrolplane.LocalDiagnosticsAccessConfiguration {
+	model := input[0]
+	output := packetcorecontrolplane.LocalDiagnosticsAccessConfiguration{
+		AuthenticationType: packetcorecontrolplane.AuthenticationType(model.AuthenticationType),
+	}
+	if model.HttpsServerCertificateUrl != "" {
+		output.HTTPSServerCertificate = &packetcorecontrolplane.HTTPSServerCertificate{
+			CertificateUrl: model.HttpsServerCertificateUrl,
+		}
+	}
+	return output
+}
+
+func flattenLocalPacketCoreControlLocalDiagnosticsAccessConfiguration(input packetcorecontrolplane.LocalDiagnosticsAccessConfiguration) []LocalDiagnosticsAccessConfigurationModel {
+	outputs := make([]LocalDiagnosticsAccessConfigurationModel, 0)
+	output := LocalDiagnosticsAccessConfigurationModel{
+		AuthenticationType: string(input.AuthenticationType),
+	}
+	if input.HTTPSServerCertificate != nil {
+		output.HttpsServerCertificateUrl = input.HTTPSServerCertificate.CertificateUrl
+	}
+	outputs = append(outputs, output)
+	return outputs
+}
+
+func expandPlatformConfigurationModel(inputList []PlatformConfigurationModel) (packetcorecontrolplane.PlatformConfiguration, error) {
+	output := packetcorecontrolplane.PlatformConfiguration{}
+	if len(inputList) == 0 {
+		return output, nil
+	}
+
+	input := inputList[0]
+
+	output.Type = packetcorecontrolplane.PlatformType(input.Type)
+
+	if pass, err := verifyPlatformConfigurationModel(input); !pass {
+		return output, err
+	}
+
+	if input.AzureStackEdgeDeviceId != "" {
+		output.AzureStackEdgeDevice = &packetcorecontrolplane.AzureStackEdgeDeviceResourceId{
+			Id: input.AzureStackEdgeDeviceId,
+		}
+	}
+
+	if input.ConnectedClusterId != "" {
+		output.ConnectedCluster = &packetcorecontrolplane.ConnectedClusterResourceId{
+			Id: input.ConnectedClusterId,
+		}
+	}
+
+	if input.AzureStackHciClusterId != "" {
+		output.AzureStackHciCluster = &packetcorecontrolplane.AzureStackHCIClusterResourceId{
+			Id: input.AzureStackHciClusterId,
+		}
+	}
+
+	if input.CustomLocationId != "" {
+		output.CustomLocation = &packetcorecontrolplane.CustomLocationResourceId{
+			Id: input.CustomLocationId,
+		}
+	}
+
+	return output, nil
+}
+
+func verifyPlatformConfigurationModel(input PlatformConfigurationModel) (bool, error) {
+	idList := make([]string, 0)
+	if input.AzureStackEdgeDeviceId != "" {
+		idList = append(idList, input.AzureStackEdgeDeviceId)
+	}
+	if input.AzureStackHciClusterId != "" {
+		idList = append(idList, input.AzureStackHciClusterId)
+	}
+	if input.ConnectedClusterId != "" {
+		idList = append(idList, input.ConnectedClusterId)
+	}
+
+	if len(idList) == 0 {
+		return false, fmt.Errorf("at least one of `arc_kubernetes_cluster_id`, `stack_hci_cluster_id` and `custom_location_id` should be specified")
+	}
```

Comment:
Not sure why we need this check here if it's already specified in the schema with `AtLeastOneOf`